solvers/python-CondatVu.py

```diff
@@ -21,7 +25,9 @@ def set_objective(self, A, reg, y):
 
     def run(self, callback):
         len_y = len(self.y)
-        D = (np.eye(len_y, k=1) - np.identity(len_y))[:-1]
+        data = np.array([np.ones(len_y), -np.ones(len_y)])
+        diags = np.array([0, 1])
+        D = spdiags(data, diags, len_y-1, len_y).toarray()
```

Comment:
keep it as a sparse matrix don't make it a numpy array
```suggestion
        D = spdiags(data, diags, len_y-1, len_y)
```

Reply:
Should I turn all the variables (like u, z, ...) to matrix? Otherwise I can't keep them the same type after the operations with D.

Reply:
no need

Reply:
If I keep D a format matrix, there are mang other function that I can no longer use, like np.sign() and np.linalg.norm().